pkg/filesystem/path/unix_parser.go

```diff
@@ -0,0 +1,129 @@
+package path
+
+import (
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type unixParser struct {
+	path string
+}
+
+type unixRelativeParser struct {
+	path string
+}
+
+func stripOneOrMoreSlashes(p string) string {
+	for {
+		p = p[1:]
+		if p == "" || p[0] != '/' {
+			return p
+		}
+	}
+}
+
+func (p *unixParser) ParseScope(scopeWalker ScopeWalker) (next ComponentWalker, remainder RelativeParser, err error) {
```

Comment:
I would change this to not use a pointer receiver. So `p unixParser`. Relatedly, I would advise against mutating `p.path`.

What are your thoughts on just writing this function like this?
```go
if p.path != "" && p.path[0] == '/' {
    next, err := scopeWalker.OnAbsolute()
    if err != nil {
        return nil, nil, err
    }
    return next, unixRelativeParser{stripOneOrMoreSlashes(p.path)}, nil
}

next, err := scopeWalker.OnRelative()
if err != nil {
    return nil, nil, err
}
return next, unixRelativeParser{p.path}, nil

Reply:
Looks good! It is annoying to try to create a variable for the strip function, as a var declaration and then use that at the very end.